TESTS/dev_mgmt/update/main.cpp

```diff
@@ -195,37 +168,32 @@ void spdmc_testsuite_update(void) {
 
     test_case_finish("Connect to " TEST_NETWORK_TYPE, iteration + (net_status == 0), (net_status != 0));
 
-    if (iteration == 0) {
-        test_case_start("Format " TEST_FILESYSTEM_TYPE, 3);
```

Comment:
The reason why I introduced the formatting in this function, is because quite often we get test failures due to the filesystem not being correctly erased. This code ensures that that we don't run into issues unrelated to the functionality of the board, mbed OS and client, i.e. the storage might have various garbage data I don't want us to test against.

Reply:
I am formatting file system at first iteration
int client_status = client.init( (iteration == 0) ? true : false ); 
it's not possible now to format before init, you don't know the partition mode.
you have to use the partition mode now...

but I am formatting SD at init, as you run first iteration, see new parameter at init function.
format ( true/false)

Reply:
should not be "garbage" problem